resources/axon.rb

```diff
@@ -98,82 +99,132 @@
     not_if { new_resource.tags.empty? }
   end
 
-  # Set local cache target for the installers
-  local_installer = ::Chef::Config['file_cache_path'] + '/te_agent' + ext
+  agent_source_is_url = new_resource.installer.start_with?('http')
+  agent_source_is_tgz = new_resource.installer.end_with?('tgz')
+  agent_source_is_zip = new_resource.installer.end_with?('zip')
 
   # Set the correct header for the agent installer
-  agent_source = if new_resource.installer.start_with?('http')
-                   new_resource.installer
-                 else
-                   'file:///' + new_resource.installer
-                 end
-
-  # Download installer
-  remote_file local_installer do
-    source agent_source
-    mode '744' unless node['platform'] == 'windows'
+  agent_source = agent_source_is_url ? new_resource.installer : 'file:///' + new_resource.installer
+  folder = ::File.basename(agent_source, '.*')
+  axon_chef_cache = "#{::Chef::Config['file_cache_path']}/#{folder}"
```

Comment:
the hardcoded file separator makes me nervous. This is evaluated at run time on the guest box? or is this evaluated at compile time?

Reply:
File.join()

I remembered after posting this that ruby uses forward slash for both linux and windows, so it's not a big deal, but File.join would probably be cleaner.